rero_ils/modules/acq_order_lines/api.py

```diff
@@ -183,6 +194,61 @@ def unreceived_quantity(self):
         """Get the number of item not yet received."""
         return calculate_unreceived_quantity(self)
 
+    @property
+    def quantity(self):
+        """Get quantity of ordered_items for a line order.
+        
+        This comes from the metadata of the order line that represent the
+        number of items to order or already ordered.
+        """
+        return self.get('quantity')
+
+    @property
+    def received_quantity(self):
+        """Get quantity of received ordered_items for a line order.
+        
+        The received quantitiy is number of quantity received for the resource
+        acq_receipt_line and for the correspoding acq_line_order.
+        """
+        from rero_ils.modules.acq_receipt_lines.api import \
+            AcqReceiptLinesSearch
+        search = AcqReceiptLinesSearch()\
+            .filter('term', acq_account__pid=self.account_pid)
+        search.aggs.metric('sum_order_line_recieved', 'sum', field='quantity')
+        results = search.execute()
+        return results.aggregations.sum_order_line_recieved.value
+
+    @property
+    def unreceived_quantity(self):
+        """Get quantity of unreceived ordered_items for a line order."""
+        return self.quantity - self.received_quantity
+
+    @property
+    def status(self):
+        """Calculate the order line status.
+
+        The status of the order line is saved only in the elasticsearch index
+        and it is calculated based on the following roles:
+        * at the creation, the order line receives the `APPROVED` status
+        * if field `cancelled` is checked, it receives the `CANCELLED`status
+        * if some items are received but not the total, it receives
+            `PARTIALLY_RECEIVED` status
+        * if all items are received, it receives `RECEIVED` status
+        """
+        received_quantity = self.received_quantity
+        unreceived_quantity = self.quantity - received_quantity
+        status = AcqOrderLineStatus.APPROVED
+        if self.cancelled:
+            status = AcqOrderLineStatus.CANCELLED
+        elif self.order_date:
+            status = AcqOrderLineStatus.ORDERED
+        elif not unreceived_quantity:
+            status = AcqOrderLineStatus.RECEIVED
+        elif unreceived_quantity and unreceived_quantity != self.quantity:
+            status = AcqOrderLineStatus.PARTIALLY_RECEIVED
+
+        return status
```

Comment:
smallest implementation :
```python
if self.is_cancelled:
    return AcqOrderLineStatus.CANCELLED
status = AcqOrderLineStatus.ORDERED if self.order_data else AcqOrderLineStatus.APPROVED
received_quantity = self.received_quantity
unreceived_quantity = self.quantity - received_quantity  # not use the property to prevent an extra ES call
if unreceived_quantity == 0:  # fully received
    status = AcqOrderLineStatus.RECEIVED
elif unreceived_quantity and unreceived_quantity != self.quantity:
    status = AcqOrderLineStatus.PARTIALLY_RECEIVED
return status
```